test/fuzzy_object_store/fuzz_engine.cpp

```diff
@@ -0,0 +1,190 @@
+/*************************************************************************
+ *
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **************************************************************************/
+
+#include "fuzz_engine.hpp"
+#include "fuzz_configurator.hpp"
+#include "fuzz_object.hpp"
+#include "util.hpp"
+
+#include <ctime>
+#include <cstdio>
+#include <fstream>
+#include <iostream>
+
+using namespace realm;
+const size_t max_tables = REALM_MAX_BPNODE_SIZE * 10;
+
+const char hex_digits[] = "0123456789abcdefgh";
+char hex_buffer[3];
+
+static const char* to_hex(char c)
+{
+    hex_buffer[0] = hex_digits[(c >> 4) & 0x0f];
+    hex_buffer[1] = hex_digits[c & 0x0f];
+    return hex_buffer;
+}
+
+int FuzzEngine::run_fuzzer(const std::string& input, const std::string& name, bool enable_logging,
+                           const std::string& path)
+{
+    auto configure = [&](auto fuzzer) {
+        try {
+            FuzzConfigurator cnf(fuzzer, input, false, name);
+            if (enable_logging) {
+                cnf.get_logger().enable_logging(path);
+                cnf.print_cnf();
+            }
+            return cnf;
+        }
+        catch (const EndOfFile& e) {
+            std::cout << "Error, cnf is invalid" << std::endl;
+            throw std::runtime_error{"Realm cnf is invalid"};
+        }
+    };
+
+    try {
+        FuzzObject fuzzer;
+        auto cnf = configure(fuzzer);
+        do_fuzz(cnf);
+    }
+    catch (const EndOfFile&) {
+        std::cout << "End of input ... " << std::endl;
```

Comment:
I don't think this should be written to stdout. It doesn't really add any value.

Reply:
done